tests/elasticity_test.py

```diff
@@ -0,0 +1,487 @@
+      atol = 1e-8
+      rtol = 1e-5
+
+    for index in range(STOCHASTIC_SAMPLES):
+      cijkl, R, sigma, box = test_util.load_nc_data(spatial_dimension, low_pressure, dtype, index) 
+      R = space.transform(box, R)
+      box = box[0,0]
+
+      displacement, shift = space.periodic(box)
+      energy_fn = energy.soft_sphere_pair(displacement, sigma=1.0) #This is the wrong sigma, so must pass dynamically
+      assert( jnp.max(jnp.abs(grad(energy_fn)(R, sigma=sigma))) < max_grad_thresh )
+
+      EMT_fn = jit(elasticity.AthermalElasticModulusTensor(energy_fn, check_convergence=True))
+      C, converged = EMT_fn(R,box,sigma=sigma)
+      assert( C.dtype == dtype )
+      assert( C.shape == (spatial_dimension,spatial_dimension,spatial_dimension,spatial_dimension) )
+      if converged:
+        self.assertAllClose(cijkl,elasticity._extract_elements(C,False), atol=atol, rtol=rtol)
+        
+        #make sure the symmetries are there
+        self.assertAllClose(C, jnp.einsum("ijkl->jikl", C))
+        self.assertAllClose(C, jnp.einsum("ijkl->ijlk", C))
+        self.assertAllClose(C, jnp.einsum("ijkl->lkij", C))
+
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {
+          'testcase_name': '_dim={}_dtype={}_lowpressure={}'.format(dim, dtype.__name__, low_pressure),
+          'spatial_dimension': dim,
+          'dtype': dtype,
+          'low_pressure': low_pressure
+      } for dim in SPATIAL_DIMENSION for dtype in DTYPE for low_pressure in LOWPRESSURE))
+  def test_EMT_from_database_nbrlist(self, spatial_dimension, dtype, low_pressure):
+    if spatial_dimension == 2:
+      N = 64
+    else:
+      N = 128
+
+    if dtype == jnp.float32:
+      max_grad_thresh = 1e-5
+      atol = 1e-4
+      rtol = 1e-3
+    else:
+      max_grad_thresh = 1e-10
+      atol = 1e-8
+      rtol = 1e-5
+
+    for index in range(STOCHASTIC_SAMPLES):
+      cijkl, R, sigma, box = test_util.load_nc_data(spatial_dimension, low_pressure, dtype, index) 
+
+      displacement, shift = space.periodic_general(box, fractional_coordinates=True)
+      neighbor_fn, energy_fn = energy.soft_sphere_neighbor_list(displacement, box, sigma=sigma, fractional_coordinates=True)
+      nbrs = neighbor_fn.allocate(R)
+      assert( jnp.max(jnp.abs(grad(energy_fn)(R, nbrs))) < max_grad_thresh )
+
+      EMT_fn = jit(elasticity.AthermalElasticModulusTensor(energy_fn, check_convergence=True))
+      C, converged = EMT_fn(R,box,neighbor=nbrs)
+      assert( C.dtype == dtype )
+      assert( C.shape == (spatial_dimension,spatial_dimension,spatial_dimension,spatial_dimension) )
+      if converged:
+        self.assertAllClose(cijkl,elasticity._extract_elements(C,False), atol=atol, rtol=rtol)
+        
+        #make sure the symmetries are there
+        self.assertAllClose(C, jnp.einsum("ijkl->jikl", C))
+        self.assertAllClose(C, jnp.einsum("ijkl->ijlk", C))
+        self.assertAllClose(C, jnp.einsum("ijkl->lkij", C))
+
+
+
+      """
```

Comment:
These comments should probably either be removed or incorporated.

Reply:
Whoops, they were already incorporated, I just forgot to delete this.